### PR TITLE
Move files that are not golden files into `input` directory

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Move files that are not golden files into `input` directory
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Golden files are only those files that can be generated with `CREATE_GOLDEN_FILES=1`.

Any other files are not golden files and should go elsewhere.  For example input files should go in the `input` directory.

The benefit of this split is that you could `rm -rf` the `golden` directory knowing all the files within can be regenerated.  This is a good way to figure out if any golden files are not longer in use.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Shelley/MultiSig/Address.hs [hunk 1]
```diff
@@ -12,47 +12,47 @@ import qualified Hedgehog.Extras.Test.File as H
...
+  atLeastMultiSigFp <- noteInputFile "test/cardano-cli-golden/files/input/shelley/multisig/scripts/atleast"
 
   atLeastMultiSigAddress <- execCardanoCLI
     [ "address", "build"
     , "--payment-script-file", atLeastMultiSigFp
     , "--mainnet"
     ]
 
-  goldenAtLeastMultiSigAddrFp <- noteInputFile "test/cardano-cli-golden/files/golden/shelley/multisig/addresses/atleast"
+  goldenAtLeastMultiSigAddrFp <- noteInputFile "test/cardano-cli-golden/files/input/shelley/multisig/addresses/atleast"
```
Comment (carbolymer) [inline cardano-cli/test/cardano-cli-golden/Test/Golden/Shelley/MultiSig/Address.hs]: this is not an input, but a golden file

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Shelley/Genesis/InitialTxIn.hs [hunk 2]
```diff
@@ -12,8 +12,8 @@ import qualified Hedgehog.Extras.Test.File as H
 
 hprop_golden_shelleyGenesisInitialTxIn :: Property
 hprop_golden_shelleyGenesisInitialTxIn = propertyOnce . H.moduleWorkspace "tmp" $ \tempDir -> do
-  verificationKeyFile <- noteInputFile "test/cardano-cli-golden/files/golden/shelley/keys/genesis_verification_keys/genesis-utxo.vkey"
-  goldenUtxoHashFile <- noteInputFile "test/cardano-cli-golden/files/golden/shelley/keys/genesis_utxo_hashes/utxo_hash"
+  verificationKeyFile <- noteInputFile "test/cardano-cli-golden/files/input/shelley/keys/genesis_verification_keys/genesis-utxo.vkey"
+  goldenUtxoHashFile <- noteInputFile "test/cardano-cli-golden/files/input/shelley/keys/genesis_utxo_hashes/utxo_hash"
```
Comment (carbolymer) [inline cardano-cli/test/cardano-cli-golden/Test/Golden/Shelley/Genesis/InitialTxIn.hs]: not an input, but golden file

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/Vote.hs [hunk 3]
```diff
@@ -46,9 +46,9 @@ hprop_golden_byron_yes_vote = propertyOnce $ H.moduleWorkspace "tmp" $ \tempDir
 
 hprop_golden_byron_no_vote :: Property
 hprop_golden_byron_no_vote = propertyOnce $ H.moduleWorkspace "tmp" $ \tempDir -> do
-  goldenNoVote <- noteInputFile "test/cardano-cli-golden/files/golden/byron/votes/vote-no"
-  proposal <- noteInputFile "test/cardano-cli-golden/files/golden/byron/update-proposal"
-  signingKey <- noteInputFile "test/cardano-cli-golden/files/golden/byron/keys/byron.skey"
+  goldenNoVote <- noteInputFile "test/cardano-cli-golden/files/input/byron/votes/vote-no"
```
Comment (carbolymer) [inline cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/Vote.hs]: no an input, but golden

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/Vote.hs [hunk 4]
```diff
@@ -19,9 +19,9 @@ import           Hedgehog.Internal.Property (failWith)
 
 hprop_golden_byron_yes_vote :: Property
 hprop_golden_byron_yes_vote = propertyOnce $ H.moduleWorkspace "tmp" $ \tempDir -> do
-  goldenYesVote <- noteInputFile "test/cardano-cli-golden/files/golden/byron/votes/vote-yes"
-  proposal <- noteInputFile "test/cardano-cli-golden/files/golden/byron/update-proposal"
-  signingKey <- noteInputFile "test/cardano-cli-golden/files/golden/byron/keys/byron.skey"
+  goldenYesVote <- noteInputFile "test/cardano-cli-golden/files/input/byron/votes/vote-yes"
```
Comment (carbolymer) [inline cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/Vote.hs]: no an input,but golden

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/UpdateProposal.hs [hunk 5]
```diff
@@ -19,8 +19,8 @@ import           Hedgehog.Internal.Property (failWith)
 
 hprop_golden_byron_update_proposal :: Property
 hprop_golden_byron_update_proposal = propertyOnce $ H.moduleWorkspace "tmp" $ \tempDir -> do
-  goldenUpdateProposal <- noteInputFile "test/cardano-cli-golden/files/golden/byron/update-proposal"
-  signingKey <- noteInputFile "test/cardano-cli-golden/files/golden/byron/keys/byron.skey"
+  goldenUpdateProposal <- noteInputFile "test/cardano-cli-golden/files/input/byron/update-proposal"
```
Comment (carbolymer) [inline cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/UpdateProposal.hs]: not an input, but golden

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/Tx.hs [hunk 6]
```diff
@@ -24,8 +24,8 @@ import           Hedgehog.Internal.Property (failWith)
 
 hprop_golden_byronTx_legacy :: Property
 hprop_golden_byronTx_legacy = propertyOnce $ H.moduleWorkspace "tmp" $ \tempDir -> do
-  signingKey <- noteInputFile "test/cardano-cli-golden/files/golden/byron/keys/legacy.skey"
-  goldenTx <- noteInputFile "test/cardano-cli-golden/files/golden/byron/tx/legacy.tx"
+  signingKey <- noteInputFile "test/cardano-cli-golden/files/input/byron/keys/legacy.skey"
+  goldenTx <- noteInputFile "test/cardano-cli-golden/files/input/byron/tx/legacy.tx"
```
Comment (carbolymer) [inline cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/Tx.hs]: not an input but golden

**Comment by newhoggy** [comment]
Yeah, those are annoying.  The don't meet the criteria to be called golden (ie then don't regenerate), but they call themselves golden.

**Comment by newhoggy** [comment]
I'll see if I can convert them to actually be golden.

**Comment by newhoggy** [comment]
I fixed the ones I could and renamed the rest to not be golden.

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**